### DISCUSSION TITLE
Less memory reported than what is installed.

### DISCUSSION THREAD

**Discussion by dembeck** [memtest86plus/memtest86plus#169] [Q&A] [answered]
# Specs

  - MemTest86+ Version: v6.00b3

  - **System:**
    - Dell PowerEdge R610
    - CPU: 2 x Xeon E5620 2.4GHz (Westmere)
    - Memory: 12 x 16GB 2Rx4 PC3L 10600 Registered ECC (192GB Total)
    - Boot Mode: Legacy (UEFI Disabled)
    - Memory Watchdog: Disabled

# Issues and Questions

## Issue

MemTest86+ is reporting **191GB** of memory, not the full **192GB** that is installed and reported by the system during bootup.

  - **Is all the memory being tested?**

## Question

Also, MemTest86+ completed the first round of testing in about 9 hours and reported **PASS**. The second round of testing is taking several hours longer to complete.

  - **Is it normal for the time to complete a round of testing to differ by hours?**

## Question

Being that the installed memory is ECC, and that this provides error correction. 

  - **Is MemTest86+ able to fully detect memory errors with ECC memory?**

# Screenshot

![Screenshot 2022-09-13 150124-edit](https://user-images.githubusercontent.com/9568447/190020962-c4f699ad-f816-4252-9fbf-0281a79df59f.png)

**Comment by x86fr** [comment, marked as answer]
> Is all the memory being tested?

All _**usable**_ memory (by an operating system) is being tested. Rarely, some memory can be reserved by the motherboard for other uses, like Video RAM or as a RAID controller cache. This reserved memory can't be accessed by any software.

> Is it normal for the time to complete a round of testing to differ by hours?

Yes. The first pass is reduced to optimize the time vs error detection ratio.

> Is MemTest86+ able to fully detect memory errors with ECC memory?

Memtest86+ is able to detect memory errors (ie: uncorrectable memory errors on an ECC system). If no memory errors are found (eg: a bit flip that has been corrected on-the-fly by an error mechanism like ECC), it currently can't detect it. Corrected memory errors detection on ECC system is planned on future updates.
